core/storage/base_model/gae_models.py

```diff
@@ -1324,12 +1316,11 @@ def get_multi_versions(
 
     @classmethod
     def get(
-            cls, # type: Type[SELF_VERSIONED_MODEL]
-            entity_id, # type: Text
-            strict=True, # type: bool
-            version=None # type: Optional[int]
-    ):
-        # type: (...) -> Optional[SELF_VERSIONED_MODEL]
+            cls: Type[SELF_VERSIONED_MODEL],
```

Comment:
Why `cls` needs to be typed here?

Reply:
@vojtechjelinek This is known as self typing.
Supose there are 2 cases:
```
# Case 1:
class A:
  def get(cls, ..) -> A:  ...

# Case 2:
class A:
  def get(cls: Type[BASE_TYPE], ...) -> BASE_TYPE: ...
```
And we define class B:
```
class B(A): ...

B.get(..) # Case 1: type recognized by mypy = A 
B.get(..) # Case 2: type recognized by mypy = B
```

Self typing is used to match the types of a method based on type of self or cls. Thanks!